sbom/diff.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# This script runs sbom-generator diff against the committed SBOM.
+# The diff subcommand generates a fresh SBOM internally and compares
+# it against the existing SBOM at --output-path. Exits non-zero on
+# meaningful differences.
+
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
+
+WRAPPER="${REPO_ROOT}/scripts/run-from-container.sh"
+IMAGE="sbom-generator-test"
+CONFIG="${REPO_ROOT}/sbom/sbom.config.yaml"
+COMMITTED_SBOM="${REPO_ROOT}/sbom/sbom.cdx.json"
+
+if [[ ! -x "${WRAPPER}" ]]; then
+  echo "Missing or non-executable ${WRAPPER}." >&2
+  exit 1
+fi
+
+# Ensure cache and output directories exist
+XDG_CACHE_HOME="${REPO_ROOT}/.cache"
```

Comment:
Question: If this is suppose to be on the host (see other comment), do we want to keep the user preference if set?

```suggestion
XDG_CACHE_HOME="${XDG_CACHE_HOME:-${REPO_ROOT}/.cache}"
```

Reply:
Same as above